src/components/TodoList/TodoList.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Todo } from '../Todo/Todo';
+
+export const TodoList = ({
+  todos,
+  removeTodo,
+  toggleSetCompleted,
+  toggleSetAllCompleted,
+}) => (
+  <>
+    <input
+      type="checkbox"
+      id="toggle-all"
+      className="toggle-all"
+      onChange={evt => toggleSetAllCompleted(evt.target.checked)}
+      checked={todos.length && todos.every(todo => todo.completed)}
```

Comment:
todos.every(todo => todo.completed) - move outside of the render or [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)

Reply:
rewrited outside of the render